### ISSUE TITLE
add allowUnnamedVariables property to support Java 21+ unnamed variable

### ISSUE THREAD

**Issue by Devansh-18155** [checkstyle/checkstyle#21356]
I have read check documentation: https://checkstyle.org/checks/naming/localvariablename.html
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Java 21 introduced unnamed variables `(_).` Currently,` LocalVariableNameCheck` flags `int _ = 10; `as a naming violation because` _ `does not match the default camelCase pattern.

We propose adding a boolean property` allowUnnamedVariables` to` LocalVariableNameCheck `so that local variables named` _ `can be ignored when set to` true.`

**Comment by Devansh-18155** [comment]
i am on it if `approved`

**Comment by romani** [comment]
Share all CLI output as it was requested in issue template

**Comment by Devansh-18155** [comment]
The CLI output regarding the issue coming is :

```bash
/var/tmp $ cat InputLocalVariableNameUnnamed.java
public class InputLocalVariableNameUnnamed {
    public void testMethod() {
        int _ = 10;
    }
}

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LocalVariableName"/>
  </module>
</module>

/var/tmp $ javac InputLocalVariableNameUnnamed.java

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.18.1-all.jar -c config.xml InputLocalVariableNameUnnamed.java
Starting audit...
[ERROR] /var/tmp/InputLocalVariableNameUnnamed.java:3:13: Name '_' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
Audit done.

**Comment by romani** [comment]
>10.18.1

Please use 14.0.0

**Comment by Devansh-18155** [comment]
Updated CLI output using version `14.0.0`:

```bash
/var/tmp $ cat InputLocalVariableNameUnnamed.java
public class InputLocalVariableNameUnnamed {
    public void testMethod() {
        int _ = 10;
    }
}

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LocalVariableName"/>
  </module>
</module>

/var/tmp $ javac InputLocalVariableNameUnnamed.java

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-14.0.0-all.jar -c config.xml InputLocalVariableNameUnnamed.java
Starting audit...
[ERROR] /var/tmp/InputLocalVariableNameUnnamed.java:3:13: Name '_' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
Audit done.

**Comment by ZaheerAhmadDev** [comment]
User can do that with format property they have a choice if they do not want violation for `Unnamed` `_` variable.
see at:https://checkstyle.sourceforge.io/checks/naming/localvariablename.html

```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LocalVariableName">
      <property name="format" value="^(_|[a-z][a-zA-Z0-9]*)$"/>
    </module>
  </module>
</module>
```
```java
cat src/Test.java
public class Test {
 public void testMethod() {
  int _ = 10;
 }
}
```

```
$ java -jar checkstyle-14.0.0-all.jar -c config.xml  src/Test.java
Starting audit...
Audit done.
```

**Comment by Devansh-18155** [comment]
Yeah, I checked. Since `format` can be used via regex (`_` can be included), is a dedicated `allowUnnamedVariables` property necessary for `LocalVariableNameCheck`, or should we rely on `format` only?

**Comment by ZaheerAhmadDev** [comment]
> Yeah, I checked. Since `format` can be used via regex (`_` can be included), is a dedicated `allowUnnamedVariables` property necessary for `LocalVariableNameCheck`, or should we rely on `format` only?
> 



Format property can suppress the violation, adding a new property is unnecessary.

**Comment by Devansh-18155** [comment]
I Agreed. Since the `format` property already allows suppressing violations for unnamed `_` variables, adding a new property is unnecessary. Closing this issue.
